8-web-components/1-webcomponents-intro/article.md

```diff
@@ -1,76 +1,76 @@
+¿Cómo decidimos qué es un componente? Eso viene de la intuición, la experiencia y el sentido común. Normalmente es una entidad visual separada que podemos describir en términos de lo que hace y cómo interactua con la página. En el caso anterior, la página tiene bloques, cada uno de ellos juega su propio papel, es lógico crear esos componentes.
 
-A component has:
-- Its own JavaScript class.
-- DOM structure, managed solely by its class, outside code doesn't access it ("encapsulation" principle).
-- CSS styles, applied to the component.
-- API: events, class methods etc, to interact with other components.
+Un componentes tiene:
+- Su propia clase de JavaScript.
+- La estructura DOM, gestionada únicamente por su clase, el código externo no accede a ella (principio de  "encapsulación").
+- Estilos CSS, aplicados al componente.
+- API: eventos, métodos de clase etc, para interactuar con otros componentes.
 
-Once again, the whole "component" thing is nothing special.
+Una vez más, todo el asunto del "componente" no es nada especial.
 
-There exist many frameworks and development methodologies to build them, each with its own bells and whistles. Usually, special CSS classes and conventions are used to provide "component feel" -- CSS scoping and DOM encapsulation.
+Existen muchos frameworks y metodologías de desarrollos para construirlos, cada uno con sus propias características y reglas. Normalmente, se utilizan clases y convenciones CSS para proporcionar la "sensación de componente" -- alcance de CSS y encapsulación de DOM.
 
-"Web components" provide built-in browser capabilities for that, so we don't have to emulate them any more.
+"Web components" proporcionan capacidades de navegación incorporadas para eso, así que ya no tenemos que emularlos.
 
-- [Custom elements](https://html.spec.whatwg.org/multipage/custom-elements.html#custom-elements) -- to define custom HTML elements.
-- [Shadow DOM](https://dom.spec.whatwg.org/#shadow-trees) -- to create an internal DOM for the component, hidden from the others.
-- [CSS Scoping](https://drafts.csswg.org/css-scoping/) -- to declare styles that only apply inside the Shadow DOM of the component.
-- [Event retargeting](https://dom.spec.whatwg.org/#retarget) and other minor stuff to make custom components better fit the development.
+- [Custom elements](https://html.spec.whatwg.org/multipage/custom-elements.html#custom-elements) -- para definir elementos HTML personalizados.
+- [Shadow DOM](https://dom.spec.whatwg.org/#shadow-trees) -- para crear un DOM interno para el componente, oculto a los demás componentes.
+- [CSS Scoping](https://drafts.csswg.org/css-scoping/) -- para declarar estilos que sólo se aplican dentro del Shadow DOM del componente.
```

Comment:
Shadow DOM es una forma de encapsular elementos DOM dentro del propio DOM, como si fuese subárboles de elementos, pueden ser específicos de un componente. Podríamos dejarlo sin traducir, añadiendo alguna url de sugerencia, por ejemplo: https://www.todojs.com/shadow-dom-a-fondo/ . Aunque en mi opinión es mejor inducir a los recién llegados a la investigación, es decir, que ellos mismos busquen y entiendan que significan ciertas cosas . :hand_over_mouth:

Reply:
Seleccionar, clic secundario, "search google for..."  - Si no se aguanta la curiosidad.
Si, me parece bien.  
El sitio parece interesante, pero arranca metiéndote en las shadows, 
y aca la intención parece ser dar apenas un pantallazo.
Como quieran/quierass. Va a quedar doble aprobado, ok.